sort/pigeonhole_sort.dart

```diff
@@ -21,7 +21,7 @@ void pigeonholeSort(List arr) {
   int range = max - min;
   range++;
 
-  List phole = new List(range);
+  List phole = []..length = range;
```

Comment:
I think assigning length value is not a good practice. Can we change the constructor to List.filled()